### PR TITLE
Adds deploy script. CI artifacts. Dependencies file

### PR DESCRIPTION
**Pull request by Cyberboss**
Adds a simple file to create a work tree with only what's needed to RUN the game.

.dmb/.rsc, _maps, strings, and some icons and sounds.

Makes the Dockerfile use it

Also adds dependencies.sh which can serve as a single version source used by all CI and utils

Enables appveyor and artifacts, you can now download a .dmb deployment straight from a PR or master commit.

From #39029

### CODE REVIEW

**Comment by Cyberboss** [comment]
Travis doesn't have free artifacts. Also, cross compiling the dmb isn't
supported iirc

**Review by MrsTonedOne — APPROVED** [review]

File: tools/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+cp -r sound/instruments/* $1/sound/instruments/
+cp -r strings/* $1/strings/
+
+#remove .dm files from _maps
+
+#this regrettably doesn't work with windows find
+#find $1/_maps -name "*.dm" -type f -delete
+
+#dlls
+cp rust_g.* $1/
```
Comment (SpaceManiac) [inline tools/deploy.sh]: this will only work on Windows; fine?

Comment (MrStonedOne) [reply tools/deploy.sh]: \>`.sh` 
\>only works on windows

Comment (SpaceManiac) [reply tools/deploy.sh]: the appveyor powershell script invokes bash on this for some reason

Comment (optimumtact) [reply tools/deploy.sh]: unacceptableeeeeeeeeee

**Review by SpaceManiac — APPROVED** [review]
this might be overengineering things a bit but I also have a pet CI change I want to make so

File: tools/deploy.sh (hunk 1 above)
Comment (Cyberboss) [reply tools/deploy.sh]: We get handy packages like this now though: https://ci.appveyor.com/api/buildjobs/a9j6dqgu0xcewq9k/artifacts/deploy.zip

**Pull request merged**